libbeat/beat/beat.go

```diff
@@ -193,5 +211,22 @@ func (b *Beat) Run() {
 
 // Stop calls the beater Stop action.
 func (beat *Beat) Stop() {
+	logp.Info("Stopping Beat")
 	beat.BT.Stop()
+	close(beat.done)
+}
+
+// Reload the beat for example with a fresh configuration
+// This requires that stop and cleanup properly shuts down the beat
```

Comment:
I too think it will take some work to make each Beat reloadable without a complete process level restart. There will likely be a lot resources that are not properly shutdown or closed.

(And yes, there is a function for getting the number of goroutines -- [NumGoroutine](https://golang.org/pkg/runtime/#NumGoroutine), but I don't think we should use it here.)